packages/dds/map/src/mapIncremental.ts

```diff
@@ -0,0 +1,466 @@
+		return this.kernel.entries();
+	}
+
+	/**
+	 * The number of key/value pairs stored in the map.
+	 */
+	public get size(): number {
+		return this.kernel.size;
+	}
+
+	/**
+	 * Executes the given callback on each entry in the map.
+	 * @param callbackFn - Callback function
+	 */
+	// TODO: Use `unknown` instead (breaking change).
+
+	public forEach(callbackFn: (value: any, key: string, map: Map<string, any>) => void): void {
+		this.kernel.forEach(callbackFn);
+	}
+
+	/**
+	 * {@inheritDoc ISharedMap.get}
+	 */
+	// TODO: Use `unknown` instead (breaking change).
+
+	public get<T = any>(key: string): T | undefined {
+		return this.kernel.get<T>(key);
+	}
+
+	/**
+	 * Check if a key exists in the map.
+	 * @param key - The key to check
+	 * @returns True if the key exists, false otherwise
+	 */
+	public has(key: string): boolean {
+		return this.kernel.has(key);
+	}
+
+	/**
+	 * {@inheritDoc ISharedMap.set}
+	 */
+	public set(key: string, value: unknown): this {
+		this.kernel.set(key, value);
+		return this;
+	}
+
+	/**
+	 * Delete a key from the map.
+	 * @param key - Key to delete
+	 * @returns True if the key existed and was deleted, false if it did not exist
+	 */
+	public delete(key: string): boolean {
+		return this.kernel.delete(key);
+	}
+
+	/**
+	 * Clear all data from the map.
+	 */
+	public clear(): void {
+		this.kernel.clear();
+	}
+
+	/**
+	 * {@inheritDoc @fluidframework/shared-object-base#SharedObject.summarizeCore}
+	 * @internal
+	 */
+	protected summarizeCore(
+		serializer: IFluidSerializer,
+		telemetryContext?: ITelemetryContext,
+		incrementalContext?: IIncrementalContext,
+	): ISummaryTreeWithStats {
+		let currentSize = 0;
+		let counter = 0;
+		let headerBlob: IMapDataObjectSerializable = {};
+		const blobs: string[] = [];
+
+		const builder = new SummaryTreeBuilder();
+
+		const data = this.kernel.getSerializedStorage(serializer);
+
+		console.log(`mapIncrementalSummarizeCore: ${JSON.stringify(incrementalContext)}`);
+
+		// Partitioning algorithm:
+		// 1) Split large (over MinValueSizeSeparateSnapshotBlob = 8K) properties into their own blobs.
+		//    Naming (across snapshots) of such blob does not have to be stable across snapshots,
+		//    As de-duping process (in driver) should not care about paths, only content.
+		// 2) Split remaining properties into blobs of MaxSnapshotBlobSize (16K) size.
+		//    This process does not produce stable partitioning. This means
+		//    modification (including addition / deletion) of property can shift properties across blobs
+		//    and result in non-incremental snapshot.
+		//    This can be improved in the future, without being format breaking change, as loading sequence
+		//    loads all blobs at once and partitioning schema has no impact on that process.
+		const keysInOrder = Object.keys(data).sort();
+		for (const key of keysInOrder) {
+			const value = data[key];
+			if (value.value && value.value.length >= MinValueSizeSeparateSnapshotBlob) {
```

Comment:
For simplicity of example, I'd recommend not to care about long properties. I.e. care only about blobs